setuptools/dist.py

```diff
@@ -636,6 +637,17 @@ def dash_to_underscore_warning(self, opt, section):
                 % (opt, underscore_opt))
         return underscore_opt
 
+    def uppercase_warning(self, opt, section):
+        if section in ('metadata',) and (any(c.isupper() for c in opt)):
```

Comment:
I think it's better to use a guard-expression style similar to the method above. It allows us to dedent the whole block of code and only keep a single line indented:
```suggestion
        if section != 'metadata' or all(not c.isupper() for c in opt):
            return opt

```

Reply:
Looking closer, I realized that you don't even need that `all()` with a generator looping over vars because you can just call `str.islower()` on the whole opt name to achieve the same:
```suggestion
        if section != 'metadata' or opt.islower():
            return opt

```